src/bin/pg_autoctl/pgctl.c

```diff
@@ -199,6 +204,11 @@ config_find_pg_ctl(PostgresSetup *pgSetup)
 		{
 			char *program = pg_ctls[i];
 			char *version = pg_ctl_version(program);
+			if (version == NULL)
+			{
+				log_warn("Failed to get version info from %s --version", program);
+				continue;
```

Comment:
I think this `continue` here warrants a comment explaining what's happening. We could have been skipping those entries from the PATH search result, for instance, and decided that if there's only one entry left that's the one we're going to use. It looks logical to do that, but I think it would be the wrong thing and your implementation is the right one. A comment would avoid rediscovering the reasoning behind it every time we look at that code.